MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -192,19 +202,19 @@ open class SimulatedLocationManager: NavigationLocationManager {
         }
         // Maximum speed if we are a safe distance from the closest coordinate
         else if distance >= safeDistance {
-            return maximumSpeed
+            return previousAverageSpeed
         }
         // Base speed on previous or upcoming turn penalty
         else {
             let reversedTurnPenalty = maximumTurnPenalty - closestLocation.turnPenalty
-            return reversedTurnPenalty.scale(minimumIn: minimumTurnPenalty, maximumIn: maximumTurnPenalty, minimumOut: minimumSpeed, maximumOut: maximumSpeed)
+            return reversedTurnPenalty.scale(minimumIn: minimumTurnPenalty, maximumIn: maximumTurnPenalty)
```

Comment:
Thanks for taking a look into this. Some output after brief testing: even though we introduce previous average speed in some cases it's not taken into account correctly. E.g. here's current speed output before and after arriving to intermediate waypoint:
```
29.774327381612984
30.082914863975745
7.919688457439359
Arrival to waypoint
6.0
6.0
6.0
```

This can be reproduced for navigation on motorways where simulation speed is higher.
Same I can see for trips with intermediate waypoint on usual streets. For example:
```
7.505037094946138
7.505037094946138
Arrival to waypoint
18.037360014387076
16.64236257388849
```

To fine tune average speed calculation algorithm we should in various cases where speed can be either really low or really fast. Ideally after arriving to intermediate waypoint current speed should be _almost_ the same as the previous one.

Reply:
I see. The first situation that caused the `6.0` after arrival to waypoint is caused by the `minimSpeed` returned when `coordinatesNearby.count >= 10`, I'll update this part. I'll revise the `calculateCurrentSpeed` to give out the similar previous speed.